File: src/robot_ipc.py
# robot_ipc.py
# Inter-process communication for RoboCode extension
from queue import Queue

class RobotIPC:
    def __init__(self):
        self.messages = Queue()
    def send_message(self, msg):
        self.messages.put(msg)
        print(f"[IPC] Sent: {msg}")
    def receive_message(self):
        if self.messages.empty():
            return ""
        msg = self.messages.get()
        print(f"[IPC] Received: {msg}")
        return msg

File: src/robot_ipc.cpp
// robot_ipc.cpp
// Inter-process communication for RoboCode extension
#include <iostream>
#include <string>
#include <thread>
#include <mutex>
#include <queue>

class RobotIPC {
public:
    void sendMessage(const std::string& msg) {
        std::lock_guard<std::mutex> lock(mtx);
        messages.push(msg);
        std::cout << "[IPC] Sent: " << msg << std::endl;
    }
    std::string receiveMessage() {
        std::lock_guard<std::mutex> lock(mtx);
        if (messages.empty()) return "";
        std::string msg = messages.front();
        messages.pop();
        std::cout << "[IPC] Received: " << msg << std::endl;
        return msg;
    }
private:
    std::queue<std::string> messages;
    std::mutex mtx;
};
